src/RobotModelUpdate.cpp

```diff
@@ -36,20 +47,123 @@ void RobotModelUpdate::reset(mc_control::MCGlobalController & controller)
+  auto RAnkle_0_W = ctl.datastore().call<sva::PTransformd>("XsensPlugin::GetSegmentPose",
+                                                           static_cast<const std::string &>("Right Foot"));
+
+  // step 2: convert world poses to local frames (kinematic tree from hips link)
+  auto Head_Hips = Head_0_W * Hips_W_0;
+  Head_Hips.translation().x() += -0.04; // offset from middle of body to actual joint
+  Head_Hips.translation().y() = 0; // center
+
+  auto LArm_Hips = LArm_W * Hips_W_0;
+  // Upper arm model is ~2/3 shirt and ~1/3 upper arm to the elbow:
+  auto LForearm_Arm = LForearm_W * LArm_W.inv();
+  LForearm_Arm.translation().x() += -0.03; // sensor is front of the arm
+  Eigen::Vector3d LElbow_Arm = 0.66 * LForearm_Arm.translation();
+  Eigen::Vector3d LForearm_Elbow = 0.33 * LForearm_Arm.translation();
+  auto LWrist_Forearm = LWrist_0_W * LForearm_W.inv();
+
+  auto RArm_Hips = RArm_W * Hips_W_0;
+  // Upper arm model is ~2/3 shirt and ~1/3 upper arm to the elbow:
+  auto RForearm_Arm = RForearm_W * RArm_W.inv();
+  RForearm_Arm.translation().x() += -0.03; // sensor is front of the arm
+  Eigen::Vector3d RElbow_Arm = 0.66 * RForearm_Arm.translation();
+  Eigen::Vector3d RForearm_Elbow = 0.33 * RForearm_Arm.translation();
+  auto RWrist_Forearm = RWrist_0_W * RForearm_W.inv();
+
+  auto LLeg_Hips = LLeg_0_W * Hips_W_0;
+  auto LShin_Leg = LShin_0_W * LLeg_0_W.inv();
+  auto LAnkle_Shin = LAnkle_0_W * LShin_0_W.inv();
+
+  auto RLeg_Hips = RLeg_0_W * Hips_W_0;
+  auto RShin_Leg = RShin_0_W * RLeg_0_W.inv();
+  auto RAnkle_Shin = RAnkle_0_W * RShin_0_W.inv();
+
+  // step 3: update joint configs using local translations
+
+  std::vector<RobotUpdateBody> joints;
+
+  joints.push_back(RobotUpdateBody{"Head_0", Head_Hips.translation()});
+
+  // joints.push_back(RobotUpdateBody{"Torso_0", Hips_W_0.inv().translation()});
+
+  joints.push_back(RobotUpdateBody{"LArm_0", LArm_Hips.translation()});
+  joints.push_back(RobotUpdateBody{"LElbow", LElbow_Arm});
+  joints.push_back(RobotUpdateBody{"LForearm", LForearm_Elbow});
+  joints.push_back(RobotUpdateBody{"LWrist_0", LWrist_Forearm.translation()});
+  joints.push_back(RobotUpdateBody{"RArm_0", RArm_Hips.translation()});
+  joints.push_back(RobotUpdateBody{"RElbow", RElbow_Arm});
+  joints.push_back(RobotUpdateBody{"RForearm", RForearm_Elbow});
+  joints.push_back(RobotUpdateBody{"RWrist_0", RWrist_Forearm.translation()});
+
+  joints.push_back(RobotUpdateBody{"LLeg_0", LLeg_Hips.translation()});
+  joints.push_back(RobotUpdateBody{"LShin_0", LShin_Leg.translation()});
+  joints.push_back(RobotUpdateBody{"LAnkle_0", LAnkle_Shin.translation()});
+  joints.push_back(RobotUpdateBody{"RLeg_0", RLeg_Hips.translation()});
+  joints.push_back(RobotUpdateBody{"RShin_0", RShin_Leg.translation()});
+  joints.push_back(RobotUpdateBody{"RAnkle_0", RAnkle_Shin.translation()});
+
+  // step 4: load configuration
+  mc_rtc::Configuration conf;
+  conf.add("joints", joints);
+  robotUpdate.load(conf);
```

Comment:
You don't need to convert to/from an `mc_rtc::Configuration` here.

```suggestion
  robotUpdate.joints = joints;
```

Or even better you can avoid the `joints` temporary alltogether and do
```cpp
auto & joints = robotUpdate.joints;
joints.clear();
joints.emplace_back("Head_0", Head_Hips.translation())
...
```

Or even even better, we could modify the schema such that `RobotUpdate` stores an `std::map<std::string, RobotUpdateBody>` instead, and avoid having to clear and re-push everything to the vector each time